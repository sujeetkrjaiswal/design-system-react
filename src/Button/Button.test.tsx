import React from 'react';
import { shallow } from 'enzyme';

import ButtonComponent from './Button';

describe('Button Component', () => {
  let props: any;

  beforeEach(() => {
    props = {
      theme: 'primary',
    };
  });

  const renderWrapper = () => shallow(<ButtonComponent {...props} />);

  describe('Snapshots', () => {
    it('should match snapshots as primary themed', () => {
      expect(renderWrapper()).toMatchSnapshot();
    });

    it('should match snapshots as secondary themed', () => {
      props.theme = 'secondary';
      expect(renderWrapper()).toMatchSnapshot();
    });
  });
});
