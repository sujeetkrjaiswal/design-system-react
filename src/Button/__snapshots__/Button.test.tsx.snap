// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshots should match snapshots as primary themed 1`] = `
<button
  className="button basic primary md"
  disabled={false}
>
  Placeholder
</button>
`;

exports[`Button Component Snapshots should match snapshots as secondary themed 1`] = `
<button
  className="button basic primary md"
  disabled={false}
>
  Placeholder
</button>
`;
